api/service/version_endpoint_service.go

```diff
@@ -123,6 +123,9 @@ func (k *endpointService) DeployEndpoint(ctx context.Context, environment *model
 		return nil, err
 	}
 
+	// Increment endpoint revision id
+	endpoint.RevisionID++
```

Comment:
Not necessarily. I expected the revision id already be incremented when we saved it to DB because the revision that failed is actually the incremented one.

Reply:
> endpoint's revisionID and any other configs from the newEndpoint should only be updated and saved to the DB if tobeDeployedEndpoint has successfully deployed.

Just to confirm my understanding, do you mean that this logic should be moved to `Deploy` in `api/queue/work/model_service_deployment.go` instead?

Reply:
Potentially, yes.

Reply:
For Merlin, we still need to save the endpoint to DB before doing the deployment because of our queue-worker implementation. In ModelServiceDeployment.Deploy(), Merlin calls the DB to get the latest endpoint data to be deployed in case other workers (other Merlin replicas) already picked up the deployment process.

I understand your concern, and I have updated the ModelServiceDeployment.Deploy() implementation to not update endpoint status from running/serving to fail if the new redeployment fails.